MekHQ/src/mekhq/campaign/mission/AtBDynamicScenario.java

```diff
@@ -294,8 +311,45 @@ public void setScenarioModifiers(List<AtBScenarioModifier> scenarioModifiers) {
         return scenarioModifiers;
     }
 
+    /**
+     * Adds a scenario modifier and any linked modifiers to this scenario,
+     * provided that the modifier exists and can be applied to the scenario (e.g. ground units on air map)
+     */
     public void addScenarioModifier(AtBScenarioModifier modifier) {
+        if (modifier == null) {
+            return;
+        }
+        
+        if ((modifier.getAllowedMapLocations() != null) && !modifier.getAllowedMapLocations().isEmpty() &&
```

Comment:
The isEmpty check is superfluous, contains below will return immediately (unless of course `getTemplate()` is expensive and you'd like to avoid that call (but I don't think it is).